src/cc_catalog_airflow/dags/provider_api_scripts/nypl.py

```diff
@@ -0,0 +1,250 @@
+from urllib.parse import urlparse, parse_qs
+from common.requester import DelayedRequester
+from common.storage.image import ImageStore
+
+logging.basicConfig(
+    format='%(asctime)s - %(name)s - %(levelname)s:  %(message)s',
+    level=logging.INFO
+)
+logger = logging.getLogger(__name__)
+
+LIMIT = 500
+DELAY = 1.0
+RETRIES = 3
+PROVIDER = "nypl"
+BASE_ENDPOINT = "http://api.repo.nypl.org/api/v1/items/search"
+METADATA_ENDPOINT = "http://api.repo.nypl.org/api/v1/items/item_details/"
+NYPL_API = os.getenv("NYPL_API_KEY")
+TOKEN = f"Token token={NYPL_API}"
+
+delay_request = DelayedRequester(delay=DELAY)
+image_store = ImageStore(provider=PROVIDER)
+
+DEFAULT_QUERY_PARAM = {
+    "q": "CC_0",
+    "field": "use_rtxt_s",
+    "page": 1,
+    "per_page": LIMIT
+}
+
+HEADERS = {
+    "Authorization": TOKEN
+}
+
+IMAGE_URL_DIMENSIONS = [
+    "g", "v", "q", "w", "r"
+]
+
+THUMBNAIL_DIMENSIONS = [
+    "w", "r", "q", "f", "v", "g"
+]
+
+
+def main():
+    page = 1
+    condition = True
+    while condition:
+        query_param = _get_query_param(
+            page=page
+        )
+        request_response = _request_handler(
+            params=query_param
+        )
+        results = request_response.get("result")
+        if type(results) == list and len(results) > 0:
+            _handle_results(results)
+            logger.info(f"{image_store.total_images} images till now")
+            page = page + 1
+        else:
+            condition = False
+    image_store.commit()
+    logger.info(f"total images {image_store.total_images}")
+
+
+def _get_query_param(
+        default_query_param=DEFAULT_QUERY_PARAM,
+        page=1,
+        ):
+    query_param = default_query_param
+    query_param["page"] = page
+    return query_param
+
+
+def _request_handler(
+        endpoint=BASE_ENDPOINT,
+        params=None,
+        headers=HEADERS,
+        retries=RETRIES
+        ):
+    results = None
+    for retry in range(retries):
+        response = delay_request.get(
+            endpoint,
+            params=params,
+            headers=headers
+        )
+        if response.status_code == 200:
+            try:
+                response_json = response.json()
+                response_json = response_json.get("nyplAPI")
+                results = response_json.get("response")
+                break
+
+            except Exception as e:
+                logger.warning(f"Request failed due to {e}")
+                results = None
+        else:
+            results = None
+    return results
+
+
+def _handle_results(results):
+    image_count = 0
+    for item in results:
+        uuid = item.get("uuid")
+
+        item_details = _request_handler(
+            endpoint=METADATA_ENDPOINT+uuid,
+        )
+        if item_details is None:
+            continue
+
+        mods = item_details.get("mods")
+        title = _get_title(
+            mods.get("titleInfo")
+        )
+        creator = _get_creators(
+            mods.get("")
+        )
```

Comment:
I think this is a bug.  It prevents ever finding creator info.

Reply:
I didn't look carefully whether creator was ingested into the tsv or not .